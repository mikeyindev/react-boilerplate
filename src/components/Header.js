import React from "react";
import { Link } from 'react-router-dom';
import { connect } from 'react-redux';
import { startLogout } from '../actions/authActions';

export const Header = (props) => (
  <header className="Header">
    <div className="Header__wrapper">
      <Link to="/dashboard">
        <h2 className="Header__title">Boilerplate</h2>
        </Link>
      <button className="Header__button--logout" onClick={props.startLogout}>
        Logout
      </button>
    </div>
  </header>
);

const mapDispatchToProps = (dispatch) => ({
  startLogout: () => dispatch(startLogout())
});

export default connect(undefined, mapDispatchToProps)(Header);
