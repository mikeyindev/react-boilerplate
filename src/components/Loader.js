import React from 'react';

const LoadingPage = () => (
  <div className="Loader">
    <img className="Loader__image" src="/images/loader.gif" />
  </div>
);

export default LoadingPage;