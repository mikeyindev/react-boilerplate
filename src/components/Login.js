import React from 'react';
import { connect } from 'react-redux';
import { startLogin } from '../actions/authActions';

export const LoginPage = (props) => <div className="Login">
    <div className="Login__box">
      <h1 className="Login__box__title">Boilerplate</h1>
      <p className="Login__box__subtitle">
        Tagline
      </p>
      <button className="Login__button" onClick={props.startLogin}>
        <span className="google-logo">
        <img src="/images/google_logo.svg" height="20px" width="20px" />
      </span>
      <span>Google Login</span>
      </button>
    </div>
  </div>;

const mapDispatchToProps = (dispatch) => ({
    startLogin: () => dispatch(startLogin())
  });

export default connect(undefined, mapDispatchToProps)(LoginPage);