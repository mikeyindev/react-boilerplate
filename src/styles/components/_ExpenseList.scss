.ExpenseList__table {
  margin: 0 auto;
  margin-top: $l-size;
  max-width: 80rem;
  width: 100%;
}

.ExpenseList__table__header {
  background: $table-header-bg;
  border: 1px solid $border-color;
  display: flex;
  justify-content: space-between;
  padding: $s-size $l-size;
}

.ExpenseListItem {
  border: 1px solid $border-color;
  border-top: none;
  color: $default-font-color;
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  padding: $s-size $l-size;
  text-decoration: none;
  transition: background 0.3s ease;
  &:hover {
    background: #EDF8F1
  }
  @media (min-width: $desktop-breakpoint) {
    align-items: center;
    flex-direction: row;
    justify-content: space-between;
  }
}

.ExpenseListItem__body {
  margin-bottom: $m-size;
  @media (min-width: $desktop-breakpoint) {
    margin-bottom: $l-size;
  }
}

.ExpenseListItem__title {
  margin: 0;
  word-break: break-all;
}

.ExpenseListItem__subtitle {
  color: $secondary-font-color;
  font-size: $s-font;
}

.ExpenseListItem__data {
  margin: $s-size 0 0 0;
  @media (min-width: $desktop-breakpoint) {
    margin: 0;
    padding-left: $m-size;
  }
}

.ExpenseListItem--message {
  align-items: center;
  justify-content: center;
  padding: $m-size;
  // Remove row highlighting from hover.
  &:hover {
    background: none;
  }
}

// Remove bold font-weight from table headers.
th {
  font-weight: normal;
}

tr {
  width: 100%;
}