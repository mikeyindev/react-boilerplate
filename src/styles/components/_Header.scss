.Header {
  background-color: $header;
  border-bottom: 1px solid $border-color;
  // color, offset-x, offset-y, blur-radius, spread-radius
  box-shadow: rgba(0, 0, 0, 0.06) 0 1px 5px 1px;
  position: fixed; // Positioned relative to the browser window
  line-height: 1;
  top: 0px; // Fix the header bar to the top
  width: 100vw;
}

.Header__button--logout {
  background: transparent;
  border: none;
  border-radius: 0px;
  color: $default-font-color;
  margin-right: $l-size;
}

.Header__button--logout:hover {
  // border-bottom: 1px solid $border-color;
  box-shadow: 0 4px 2px -3px rgba(0, 0, 0, 0.5);
}

.Header__title {
  color: $night-rider;
  h1 {
    margin: 0;
  }
}

.Header__wrapper {
  // align-items: center;
  display: flex;
  justify-content: space-between;
  margin: 0;
  // margin: 0 auto;
  padding: 0 $m-size;
  // max-width: 80rem;
  a {
    text-decoration: none;
  }
}
