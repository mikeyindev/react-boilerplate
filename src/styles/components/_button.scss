// Default styling for buttons
.button {
  background: $primary-btn-color;
  border: none;
  border-radius: 5px;
  color: white;
  display: inline-block; // Allows top & bottom margins/paddings to be respected
  font-size: $l-font;
  font-weight: 300;
  line-height: 1;
  padding: $s-size;
  text-decoration: none;
}

.button:hover {
  background: darken($primary-btn-color, 2%);
}