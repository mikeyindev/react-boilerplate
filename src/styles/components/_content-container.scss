.content-container {
  border-bottom: 1px solid $border-color;
  margin: 0 auto;
  // padding: 0 $m-size;
  margin-top: 80px; // So the fixed header bar won't conceal the content
  max-width: 80rem;
  z-index: -1;
}