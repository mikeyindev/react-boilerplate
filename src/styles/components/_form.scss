// Styling for both AddExpenseForm and EditExpenseForm
// .form {
//   display: flex;
//   flex-direction: column;
// }

.button-group {
  align-items: flex-start;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: flex-end;
  margin: $l-size 0;
}

.button--remove {
  border: none;
  color: $login-btn-color;
  font-size: $l-font;
  margin-top: $l-size;
  text-decoration: underline;
}

.button--remove:hover {
  color: darken($login-btn-color, 20%);
}

// Doing this prevents calendar from showing up  
// .DateRangePicker, .SingleDatePicker {
//   position: relative;
//   z-index: -1;
// }

.form__error {
  font-style: italic;
  margin-bottom: $m-size; 
}

@media (min-width: $desktop-breakpoint) {
  .button-group {
    flex-direction: row;  
  }

  .button--remove {
    margin: 0 0 0 $m-size;
  }
}