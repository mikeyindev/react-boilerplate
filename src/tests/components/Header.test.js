import React from 'react';
// Use shallow rendering when you don't need user interaction or lifecycle
// events
import { shallow } from 'enzyme';
// No need to import toJSON, the JSON serializer is configured in
// jest.config.json 
// import toJSON from 'enzyme-to-json';
import { Header } from '../../components/Header';

test('should render Header correctly', () => {
  const wrapper = shallow(<Header startLogout={() => { }} />);
  // expect(wrapper.find('h1').text()).toBe('Expensify');

  expect(wrapper).toMatchSnapshot();
});

test('should call startLogout on button click', () => {
  const startLogout = jest.fn();
  const wrapper = shallow(<Header startLogout={startLogout} />);
  wrapper.find('button').simulate('click');
  expect(startLogout).toHaveBeenCalled();
});