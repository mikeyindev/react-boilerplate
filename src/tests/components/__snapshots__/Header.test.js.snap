// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Header correctly 1`] = `
<header
  className="Header"
>
  <div
    className="Header__wrapper"
  >
    <Link
      replace={false}
      to="/dashboard"
    >
      <h2
        className="Header__title"
      >
        Expensify It
      </h2>
    </Link>
    <button
      className="Header__button--logout"
      onClick={[Function]}
    >
      Logout
    </button>
  </div>
</header>
`;
